tests/store/storage/test_localzarrstore.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
+import pandas as pd
+import pytest
+import xarray as xr
+import numcodecs
+from openghg.objectstore import get_writable_bucket
+from openghg.store.storage import LocalZarrStore
+from openghg.types import ZarrStoreError
+from helpers import get_footprint_datapath
+
+
+@pytest.fixture()
+def store():
+    bucket = get_writable_bucket(name="user")
+    local_store = LocalZarrStore(bucket=bucket, datasource_uuid="test-local-store-123", mode="rw")
+    yield local_store
+    local_store.delete_all()
+
+
+def test_localzarrstore_not_writable():
+    bucket = get_writable_bucket(name="user")
+    local_store = LocalZarrStore(bucket=bucket, datasource_uuid="test-local-store-123", mode="r")
+
+    with pytest.raises(PermissionError):
+        local_store.add(version="v0", dataset=xr.Dataset())
+
+    with pytest.raises(PermissionError):
+        local_store.update(version="v0", dataset=xr.Dataset())
+
+    with pytest.raises(PermissionError):
+        local_store.delete_version(version="v0")
+
+    with pytest.raises(PermissionError):
+        local_store.delete_all()
+
+
+def test_localzarrstore_add_retrieve(store):
+    datapath = get_footprint_datapath("TAC-100magl_UKV_co2_TEST_201407.nc")
+    with xr.open_dataset(datapath) as ds:
+        store.add(version="v0", dataset=ds)
+        retrieved = store.get(version="v0")
+        retrieved = retrieved.compute()
+        assert ds.equals(retrieved)
+
+    assert store.version_exists(version="v0")
+
+
+def test_dataset_retrieved_single_dataset_identical(store):
+    datapath = get_footprint_datapath("TAC-100magl_UKV_co2_TEST_201407.nc")
+    with xr.open_dataset(datapath) as ds:
+        store.add(version="v0", dataset=ds)
+
+        retrieved = store.get(version="v0")
+        assert ds.identical(retrieved)
+
+
+def test_localzarrstore_add_successive_dates_append():
+    bucket = get_writable_bucket(name="user")
+    store = LocalZarrStore(bucket=bucket, datasource_uuid="888-888", mode="rw")
+
+    file_a = get_footprint_datapath("TAC-100magl_UKV_TEST_201607.nc")
+    file_b = get_footprint_datapath("TAC-100magl_UKV_TEST_201608.nc")
+    with xr.open_dataset(file_a) as ds, xr.open_dataset(file_b) as ds2:
+        store.add(version="v0", dataset=ds)
+        store.add(version="v0", dataset=ds2)
+
+        assert store.version_exists(version="v0")
+
+        concat = xr.concat([ds, ds2], dim="time")
+        retrieved = store.get(version="v0").compute()
+
+        assert retrieved.equals(concat)
+        for var in retrieved.data_vars:
+            assert not np.sum(np.isnan(retrieved[var]))
+
+
+@pytest.mark.xfail(reason="Attributes of Datasets appended to zarr store may currently be overwritten")
```

Comment:
Does this test need to be updated or is this functionality which should be checked? Can add a specific issue for this if needed.

Reply:
Added an issue: #923